inference-engine/src/mkldnn_plugin/nodes/mkldnn_depth_to_space_node.cpp

```diff
@@ -0,0 +1,351 @@
+    auto dataType = MKLDNNExtensionUtils::IEPrecisionToDataType(precision);
+
+    auto srcDims = getParentEdgeAt(0)->getDims();
+    int nDims = srcDims.ToSizeVector().size();
+
+    InferenceEngine::LayerConfig config;
+    config.dynBatchSupport = true;
+    config.inConfs.resize(1);
+    config.outConfs.resize(1);
+    config.inConfs[0].inPlace = -1;
+    config.inConfs[0].constant = false;
+    config.outConfs[0].inPlace = -1;
+    config.outConfs[0].constant = false;
+
+    auto pushSupportedPrimitiveDescriptor = [&](const memory::format_tag memoryFormat) {
+        config.inConfs[0].desc = MKLDNNMemoryDesc(getParentEdgeAt(0)->getDims(), dataType, memoryFormat);
+        config.outConfs[0].desc = MKLDNNMemoryDesc(getChildEdgeAt(0)->getDims(), dataType, memoryFormat);
+        supportedPrimitiveDescriptors.push_back({config, impl_desc_type::unknown, memoryFormat});
+    };
+
+    auto canUseBlocked = [=](const size_t block) {
+        return srcDims[1] % block == 0 && (mode == Mode::BLOCKS_FIRST ? (srcDims[1] / block) % blockStep == 0 : block % blockStep == 0);
+    };
+
+    if (nDims == 4) {
+        pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::nhwc);
+        if (canUseBlocked(8))
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::nChw8c);
+        if (canUseBlocked(16))
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::nChw16c);
+    } else if (nDims == 5) {
+        pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::ndhwc);
+        if (canUseBlocked(8))
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::nCdhw8c);
+        if (canUseBlocked(16))
+            pushSupportedPrimitiveDescriptor(mkldnn::memory::format_tag::nCdhw16c);
+    }
+    pushSupportedPrimitiveDescriptor(MKLDNNMemory::GetPlainFormat(getParentEdgeAt(0)->getDims()));
+}
+
+void MKLDNNDepthToSpaceNode::createPrimitive() {
+    auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    auto &srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
+    if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
+        THROW_ERROR << "has not allocated destination memory";
+    if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
+        THROW_ERROR << "has not allocated input memory";
+    if (getSelectedPrimitiveDescriptor() == nullptr)
+        THROW_ERROR << "has unidentified preferable primitive descriptor";
+
+    SizeVector srcDims = getParentEdgeAt(0)->getBlob()->getTensorDesc().getDims();
+    SizeVector dstDims = getChildEdgeAt(0)->getBlob()->getTensorDesc().getDims();
+    prepareParams(srcDims, dstDims);
+
+    size_t nDims = srcDims.size();
+    const size_t nSpatialDims = nDims - 2;
+    const bool isBlocked = getParentEdgeAt(0)->getMemory().GetDesc().isBlockedCFormat();
+    nDims += nSpatialDims + static_cast<int>(isBlocked) + static_cast<int>(isBlocked && mode == Mode::DEPTH_FIRST);
+    const size_t lastIdx = nDims - 1;
```

Comment:
minor: nDims --> reshapedRank or smt to make code self explained.

Reply:
Done, thanks